File: chapter01/transformation/example-1.4.2.2/ex1422.ml
let ans =
  List.init 10 ~f:(fun i -> i + 1)
  |> List.filter ~f:(fun k -> List.init k ~f:(fun i -> i)
                              |> List.filter ~f:(fun j -> j*j > 2*k)
                              |> List.length >= 3)

File: chapter01/transformation/example-1.4.2.2/ex1422.hs
module Ex1422 where

condition :: Integer -> Integer -> Bool
condition k j = j*j > 2*k

hasAtLeast3 :: Integer -> Bool
hasAtLeast3 k = length (filter (condition k) [1..k]) >= 3

ans :: [Integer]
ans = filter hasAtLeast3 [1..10]